tutorials/first-steps.md

```diff
@@ -0,0 +1,397 @@
+
+----
+This function takes no argument
+and return a string 0️⃣
+----
+function hello = { # 1️⃣
+  return "Hello world!"
+}
+
+----
+This function takes one argument
+and return a string
+----
+function hello_who = |name| { # 2️⃣
+  return "Hello " + name
+}
+
+----
+This function takes two arguments
+and return a string
+----
+function hello_people = |name_1, name_2| -> "Hello " + name_1 + " and " + name_2 # 3️⃣
+
+----
+This is my main function
+----
+function main = |args| { # 4️⃣
+  println(hello())
+  println(hello_who("Bob"))
+  println(hello_people("Bob", "Bill"))
+}
+
+```
+
+> - 0️⃣ you can use multi-lines comments at the header of the functions
+> - 1️⃣ this is a parameter-less function
+> - 2️⃣ you need to use the pipe notation to pass parameter(s)
+> - 3️⃣ you can use a compact form (`|arg| -> 42`) to define a regular function (you don't need to use the return keyword)
+> - 4️⃣ you'll get:
+>    ```
+>    Hello world!
+>    Hello Bob
+>    Hello Bob and Bill
+>    ```
+
+✋ **To run it**, type this command: `golo golo --files 03-functions.golo`
+
+ℹ️ more about functions: [https://golo-lang.org/documentation/next/index.html#_functions](https://golo-lang.org/documentation/next/index.html#_functions).
+
+## Closures
+
+**From the Golo documentation**: *"Golo supports closures, which means that functions can be treated as first-class citizen. Defining a closure is straightforward as it derives from the way a function can be defined."*
+
+Create a new `04-closures.golo` file:
+
+> **04-closures.golo**
+```golo
+module hello.world
+
+function main = |args| {
+
+  let say_hello = -> println("Hello") # I know it's not pure 😝
+
+  say_hello() # 1️⃣
+
+  let say_hello_to_somebody = |name| {
+    println("Hello " + name)
+  }
+
+  say_hello_to_somebody("Bob") # 2️⃣
+
+  let sum = |a, b| -> a + b 3️⃣
+
+  println(
+    sum(10, sum(30, 2))
+  ) # 4️⃣ it will print 42
+
+}
+```
+
+> - 1️⃣ it will print `Hello`
+> - 2️⃣ it will print `Hello Bob`
+> - 3️⃣ Golo supports a compact form of closures for the cases where their body consists of a single expression
+> - 4️⃣ it will print `42`
+
+✋ **To run it**, type this command: `golo golo --files 04-closures.golo`
+
+ℹ️ more about closures: [https://golo-lang.org/documentation/next/index.html#_closures](https://golo-lang.org/documentation/next/index.html#_closures).
+
+
+> Most of the time, I say **Lambdas** instead **Closures**. Is this a mistake? [Are Java 8 Lambdas Closures?](https://www.bruceeckel.com/2015/10/17/are-java-8-lambdas-closures/).
+
+## Control flow
+
+Control flow in Golo has usual constructions (and some less ordinary, but we'll see it later).
+
+Create a new `05-control-flow.golo` file:
+
+> **05-control-flow.golo**
+```golo
+module hello.world
+
+function main = |args| {
+
+  let value = 5
+
+  if value <= 5 { # 1️⃣
+    println("value is <= 5")
+  } else {
+    println("value is > 5")
+  }
+
+  # the case construction can only be used inside a closure or a function
```

Comment:
you can use the case as a statement: 
```golo
let value = "one"
case {
    when value == "one" {
        println("ONE")
    }
    when value == "two" {
        println("TWO")
    }
    otherwise {
        println("NOT IN THE LIST")
    }
}
```

it's just a `if ... else if ... else if ... else` (compiled as such)